### ISSUE TITLE
numpy/testing/_private/utils.py causes `ValueError: invalid literal for int() with base 10: ''` 

### ISSUE THREAD

**Issue by sodul** [numpy/numpy#24074]
### Describe the issue:

We have some test code that ends up loading `numpy/testing/_private/utils.py`. Up until v1.24.4 everything works fine but with v1.25.0 we get a value error apparently when the packaging information is fetched. Oddly we could not reproduce when calling `list(sys_tags())` from the REPL.

### Reproduce the code example:

```python
In `numpy/testing/_private/utils.py`


    from packaging.tags import sys_tags
    _tags = list(sys_tags())
```

### Error message:

```shell
File "/Users/stephane/.pyenv/versions/3.11.4/lib/python3.11/site-packages/numpy/__init__.py", line 316, in __getattr__
    import numpy.testing as testing
  File "/Users/stephane/.pyenv/versions/3.11.4/lib/python3.11/site-packages/numpy/testing/__init__.py", line 11, in <module>
    from ._private.utils import *
  File "/Users/stephane/.pyenv/versions/3.11.4/lib/python3.11/site-packages/numpy/testing/_private/utils.py", line 64, in <module>
    _tags = list(sys_tags())
            ^^^^^^^^^^^^^^^^
  File "/Users/stephane/.pyenv/versions/3.11.4/lib/python3.11/site-packages/packaging/tags.py", line 536, in sys_tags
    yield from cpython_tags(warn=warn)
  File "/Users/stephane/.pyenv/versions/3.11.4/lib/python3.11/site-packages/packaging/tags.py", line 211, in cpython_tags
    platforms = list(platforms or platform_tags())
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/stephane/.pyenv/versions/3.11.4/lib/python3.11/site-packages/packaging/tags.py", line 395, in mac_platforms
    version = cast("MacVersion", tuple(map(int, version_str.split(".")[:2])))
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: ''
```

### Runtime information:

```
>>> import sys, numpy; print(numpy.__version__); print(sys.version)
1.25.0
3.11.4 (main, Jun 27 2023, 17:18:33) [Clang 14.0.3 (clang-1403.0.22.14.1)]
>>> print(numpy.show_runtime())
WARNING: `threadpoolctl` not found in system! Install it by `pip install threadpoolctl`. Once installed, try `np.show_runtime` again for more detailed build information
[{'numpy_version': '1.25.0',
  'python': '3.11.4 (main, Jun 27 2023, 17:18:33) [Clang 14.0.3 '
            '(clang-1403.0.22.14.1)]',
  'uname': uname_result(system='Darwin', node='ip-192-168-0-32.us-west-2.compute.internal', release='22.5.0', version='Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:20 PDT 2023; root:xnu-8796.121.3~7/RELEASE_ARM64_T6000', machine='arm64')},
 {'simd_extensions': {'baseline': ['NEON', 'NEON_FP16', 'NEON_VFPV4', 'ASIMD'],
                      'found': ['ASIMDHP', 'ASIMDDP'],
                      'not_found': ['ASIMDFHM']}}]
None
```

Edit: (mattip) formatting

**Comment by mattip** [comment]
This was changed in df3751b03d to add the tags code to test for musllinux, which admittedly can be fragile. I seem to recall a problem with some versions of macOS not properly adding a `.` in the version name, causing `version_str` to be something like `10` or `11`, instead of `10.1`. @andyfaff perhaps we should rethink that stanza and prefer the `sysconfig.get_config_var('HOST_GNU_TYPE')` method?

**Comment by andyfaff** [comment]
I'll report an issue upstream for this. 
From memory I don't think your suggestion will work. Hard as it is to believe I couldn't find a way to see if a distro is musl with just the standard library

**Comment by mattip** [comment]
I think `sysconfig.get_config_var("EXT_SUFFIX")` is guaranteed to always exist, and as far as I can tell always contains `musl` for musl linux.

**Comment by andyfaff** [comment]
I can't reproduce the OP issue. I have a python 3.11.4 (conda) environment on macOS, and installed `packaging-23.1`. Perhaps it's an issue with an older `packaging` version.

> I think sysconfig.get_config_var("EXT_SUFFIX") is guaranteed to always exist, and as far as I can tell always contains musl for musl linux.

Unfortunately that's not true. I started up a docker container, `quay.io/pypa/musllinux_1_1_aarch64`, and started a Python3.8 interpreter:

```
>>> import sysconfig
>>> # Doesn't contain musl, but we're in an alpine container
>>> sysconfig.get_config_var("EXT_SUFFIX")
'.cpython-38-aarch64-linux-gnu.so'
```

However, `HOST_GNU_TYPE` works in this instance:
```
>>> sysconfig.get_config_var('HOST_GNU_TYPE')
'aarch64-unknown-linux-musl'
```

See e.g. https://github.com/ekalinin/nodeenv/issues/290, for why I thought examining `HOST_GNU_TYPE` might not work comprehensively.

Anyway, perhaps it's a good idea to switch to `HOST_GNU_TYPE`. If we start getting issues with other test fails, because `IS_MUSL` isn't picked up, then we can think again. I checked with a `voidlinux-musl` and `python:3.8-alpine` and they both seemed to work.
